mozilla_version/test/test_gecko.py

```diff
@@ -326,57 +399,60 @@ def test_gecko_version_ensures_a_new_added_release_type_is_caught(monkeypatch):
     # Firefox released versions
-    '1.0rc1', '1.0rc2',
-    '1.5', '1.5rc1', '1.5rc2', '1.5rc3',
+    # TODO '0.10rc',
+    '0.10',
+    '0.10.1',
+    '0.8',
+    # TODO '0.9rc',
+    '0.9',
+    '0.9.1',
+    '0.9.2',
+    '0.9.3',
+
+    '1.0rc1', '1.0rc2', '1.0',
+    '1.0.1', '1.0.2', '1.0.3', '1.0.4', '1.0.5', '1.0.6', '1.0.7', '1.0.8',
+    '1.5b1', '1.5b2',
+    '1.5rc1', '1.5rc2', '1.5rc3', '1.5',
+    # '1.5.0.1rc1' is handled as a _RELEASED_EDGE_CASES
     '1.5.0.1',
-    '1.5.0.10',
-    '1.5.0.11',
-    '1.5.0.12',
-    '1.5.0.2',
-    '1.5.0.3',
-    '1.5.0.4',
-    '1.5.0.5',
-    '1.5.0.6',
-    '1.5.0.7',
-    '1.5.0.8',
-    '1.5.0.9',
-
-    '2.0rc1', '2.0rc2', '2.0rc3',
-    '2.0.0.1',
-    '2.0.0.3',
-    '2.0.0.4',
-    '2.0.0.5',
-    '2.0.0.6',
-    '2.0.0.7',
-    '2.0.0.8',
-    '2.0.0.9',
-    '2.0.0.2',
-    '2.0.0.10',
-    '2.0.0.11',
-    '2.0.0.12',
-    '2.0.0.13',
-    '2.0.0.14',
-    '2.0.0.15',
-    '2.0.0.16',
-    '2.0.0.17',
-    '2.0.0.18',
-    '2.0.0.19',
-    '2.0.0.20',
-
-    '3.0rc1', '3.0rc2',
-    '3.1b1',
-    '3.1b2',
-    '3.1b3',
-    '3.5b4',
+    '1.5.0.2', '1.5.0.3', '1.5.0.4', '1.5.0.5', '1.5.0.6', '1.5.0.7', '1.5.0.8', '1.5.0.9',
+    '1.5.0.10', '1.5.0.11', '1.5.0.12',
+
+    '2.0b1',
+    '2.0b2',
+    '2.0rc1', '2.0rc2', '2.0rc3', '2.0',
+    '2.0.0.1', '2.0.0.2', '2.0.0.3', '2.0.0.4', '2.0.0.5', '2.0.0.6', '2.0.0.7', '2.0.0.8',
+    '2.0.0.9', '2.0.0.10', '2.0.0.11', '2.0.0.12', '2.0.0.13', '2.0.0.14', '2.0.0.15', '2.0.0.16',
+    '2.0.0.17', '2.0.0.18', '2.0.0.19', '2.0.0.20',
+
+    '3.0rc1', '3.0rc2', '3.0rc3', '3.0',
+    '3.1b1', '3.1b2', '3.1b3', '3.5b4',
     '3.5', '3.5rc2', '3.5rc3',
     '3.6', '3.6rc1', '3.6rc2',
-    '3.6b1',
-    '3.6b2',
-    '3.6b3',
-    '3.6b4',
-    '3.6b5',
-
-    '4.0rc1', '4.0rc2',
+    '3.6b1', '3.6b2', '3.6b3', '3.6b4', '3.6b5',
+    '3.0.1', '3.0.2', '3.0.3', '3.0.4', '3.0.5', '3.0.6', '3.0.7', '3.0.8', '3.0.9', '3.0.10',
+    '3.0.11', '3.0.12', '3.0.13', '3.0.14', '3.0.15',
+    # TODO '3.0.16-real',
+    '3.0.17', '3.0.18',
+    # TODO '3.0.19-real-real',
+    '3.1b1', '3.1b2', '3.1b3',
+    '3.5b4', '3.5b99',
+    '3.5rc1', '3.5rc2', '3.5rc3', '3.5',
+    '3.5.1', '3.5.2', '3.5.3', '3.5.4', '3.5.5', '3.5.6', '3.5.7', '3.5.8', '3.5.9', '3.5.10',
+    '3.5.11', '3.5.12', '3.5.13', '3.5.14', '3.5.15', '3.5.16', '3.5.17', '3.5.18', '3.5.19',
+    '3.6rc1', '3.6rc2', '3.6',
+    '3.6b1', '3.6b2', '3.6b3', '3.6b4', '3.6b5',
+    # 3.6.1 never shipped
+    '3.6.2', '3.6.3',
+    # TODO '3.6.3plugin1',
+    '3.6.4', '3.6.6', '3.6.7', '3.6.8', '3.6.9', '3.6.10', '3.6.11', '3.6.12', '3.6.13', '3.6.14',
+    '3.6.15', '3.6.16', '3.6.17', '3.6.18', '3.6.19', '3.6.20', '3.6.21', '3.6.22', '3.6.23',
+    '3.6.24', '3.6.25', '3.6.26', '3.6.27', '3.6.28',
+
+    '4.0b1', '4.0b2', '4.0b3', '4.0b4', '4.0b5', '4.0b6', '4.0b7', '4.0b8', '4.0b9', '4.0b10',
+    '4.0b11', '4.0b12',
+    '4.0rc1', '4.0rc2', '4.0',
+    '4.0.1',
```

Comment:
Question: Hm, is this a comprehensive list of actual release numbers that we shipped, before we switched to fast-schema release model?

Reply:
Yes, I intend this list to be complete. So far, I used the data from ship-it v1, product-details, and archive.mozilla.org. I hope I didn't miss anything

Reply:
I think this is a great idea, good job for this effort! 👍  With all previous releases in this list, we can always assume we're backward compatible, if tests pass!